jgrapht-core/src/main/java/org/jgrapht/generate/PlantedPartitionGraphGenerator.java

```diff
@@ -0,0 +1,258 @@
+/*
+ * (C) Copyright 2018-2018, by Emilio Cruciani and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+
+package org.jgrapht.generate;
+
+import org.jgrapht.Graph;
+import org.jgrapht.VertexFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+
+/**
+ * Create a random $l$-planted partition graph.
+ * An $l$-planted partition graph is a random graph on $n = l \cdot k$ vertices
+ * subdivided in $l$ groups with $k$ vertices each.
+ * Vertices within the same group are connected by an edge with probability $p$,
+ * while vertices belonging to different groups are connected by an edge with probability $q$.
+ *
+ * <p>
+ * The $l$-planted partition model is a special case of the
```

Comment:
According to wiki:

> If the probability matrix is a constant, in the sense that {\displaystyle P_{ij}=p} P_{ij} = p for all {\displaystyle i,j} i,j, then the result is the Erdős–Rényi model {\displaystyle G(n,p)} G(n,p). This case is degenerate—the partition into communities becomes irrelevant—but it illustrates a close relationship to the Erdős–Rényi model.

It would perhaps be good to mention this as well, and include a reference to our G(n,p) generator which implements the Erdős–Rényi model. This makes it easier for the user to find different implementations.

Reply:
Please add the text I cited from the wiki too (without the wiki formatting)